src/wmtk/TriMeshOperationExecutor.cpp

```diff
@@ -367,6 +367,101 @@ void TriMesh::TriMeshOperationExecutor::replace_incident_face(
 }
 
 Tuple TriMesh::TriMeshOperationExecutor::split_edge()
+{
+    if (!m_mesh.multi_mesh_manager.is_parent_mesh()) {
+        return split_edge_single_mesh();
+    }
+    else
+    {
+        bool is_boundary_on_parent = m_mesh.is_boundary(m_operating_tuple);
+        
+        std::vector<Tuple> vec_t_child;
+        std::vector<Tuple> vec_t_opp_child;
+        vec_t_child = MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, m_operating_tuple);
+        if (!is_boundary_on_parent)
```

Comment:
i'd much prefer a "gather every version of this face" rather than this boundary searching logic - it doesn't generalize to other cases AND it makes the implementation much more complex. You should just be doing a for loop over a vector containing the various veresion of an edge rather than passing around two variables and copy pasting the logic for both cases everywhere. that's just asking for bugs